### ISSUE TITLE
Doesn't work for Linux

### ISSUE THREAD

**Issue by aknight2015** [Wargus/war1gus#414]
Reguardless of the version of Warcraft 1 and 2 used, it crashes with "An error with quotes has occured." I've used the GoG installer, crashes with error. Physical disk, error. Pointing to the data.war file, error. Pop!_OS (Ubuntu Distro) is the platform. All those version I tried worked with Lutris just fine, so the data files are fine. I want to use War1gus because of the updated features.

**Comment by timfel** [comment]
Please try to run the extractor on the console and post console output.

**Comment by timfel** [comment]
Also, please refer to the first minutes of https://youtu.be/6a4dy16q49c that show how to extract and what problems can occur.

**Comment by aknight2015** [comment]
So, if I try to run it from the desktop it extracts all files into the home directory, which causes the error. You gave me the idea to change to a different director (Downloads) in the console, and then run it. War1gus works fine from command. Everything extracted perfectly. Wargus (Warcraft 2) doesn't. Did the same thing for extraction for 2, but when it prompts for the expansion data or GOG installer, it doesn't recognize the Battle.net version GOG installer. I point to the Battle.net GoG installer, and it does the base game extraction all over again, then prompts again. But I'm thrilled to find out, it's a matter of permissions for the Home directory. Perhaps you should implement a way for the user to determine where the temporarily extracted files are stored. Also, please don't close this issue. Lets try and get War1gus solved first.

**Comment by aknight2015** [comment]
Yeah. I figured that out when I tested it on a whim. But I stand strong by the the fact the installer should prompt you where you want the files extracted. It helps keep system files safe, and helps with clean up as the installer doesn't clean up the files.

**Comment by timfel** [comment]
I'm not sure I understand - the innoextract files are extracted to [std::filesystem::temp_directory_path](https://en.cppreference.com/w/cpp/filesystem/temp_directory_path), which should be in `/tmp`. The converted game files are stored to `.stratagus/data.wargus`, which they have to be, since that is what the launcher uses and there's no other configuration. If you want more control over these things, you can already simply use the `wartool` and `stratagus` binaries directly. Am I misunderstanding something?

**Comment by aknight2015** [comment]
They aren't. When I use the desktop shortcut it extracts everything into the Home directory. When I used the command line everything was extracted into whatever directory I was in. So if I was in the Downloads directory in terminal, it would extract them in terminal.

**Comment by timfel** [comment]
Huh. Ok, I'll check and make sure it uses /tmp or whatever $TMPDIR is.

**Comment by aknight2015** [comment]
What I'm saying is, when I extracted to any other folder other than Home, everything went perfectly.

**Comment by timfel** [comment]
> What I'm saying is, when I extracted to any other folder other than Home, everything went perfectly.

I still don't fully get it. You literally just did this?
```bash
$ cd ~/Downoads
$ wargus
```

**Comment by aknight2015** [comment]
Yes, in terminal. When I didn't use terminal, I used the .desktop icon, it extracted everything to the Home folder. It doesn't make sense to me either, but that's what happened.
